### PR TITLE
Add Foreman package to comps and whitelist for el8

### PR DESCRIPTION
**Pull request by zjhuntin**

### CODE REVIEW

File: rel-eng/tito.props [hunk 1]
```diff
@@ -24,6 +24,7 @@ whitelist = foreman-bootloaders-redhat
   katello-certs-tools
   katello-selinux
   keycloak-httpd-client-install
+  nodejs-node-gyp
```
Comment (ekohl) [inline rel-eng/tito.props:27]: Why is this here?

File: rel-eng/tito.props [hunk 2]
```diff
@@ -351,7 +352,13 @@ scl = tfm
...
 [foreman-nightly-el8]
 disttag = .el8
-whitelist = gyp
+whitelist = foreman
+  foreman-bootloaders-redhat
+  foreman-proxy
+  foreman-release
+  foreman-release-scl
+  gyp        
+  katello-certs-tools
```
Comment (ekohl) [inline rel-eng/tito.props:361]: Duplicated line

File: rel-eng/tito.props [hunk 3]
```diff
@@ -398,9 +404,10 @@ whitelist = gyp
   nodejs-jquery
   nodejs-js-tokens
   nodejs-lodash
-  nodejs-lodash.debounce
   nodejs-lodash-es
+  nodejs-lodash.debounce
```
Comment (ekohl) [inline rel-eng/tito.props:408]: For historical reasons we sort with en_GB:
https://github.com/theforeman/foreman-packaging/blob/763e83c87d9a6d0f73177163dc2ea47298ed1506/add_gem_package.sh#L52-L61

Please keep it sorted consistently so we don't go back and forth when the script is ran again.

File: rel-eng/tito.props [hunk 4]
```diff
@@ -460,62 +467,238 @@ whitelist = gyp
...
   rubygem-coffee-rails
   rubygem-coffee-script
   rubygem-coffee-script-source
   rubygem-concurrent-ruby
-  rubygem-clamp
+  rubygem-concurrent-ruby-edge
+  rubygem-connection_pool
   rubygem-crass
+  rubygem-css_parser
+  rubygem-css_parser
```
Comment (ekohl) [inline rel-eng/tito.props:500]: Lots of duplicated entries here. The linked script above sorts and filters out duplicates.

File: comps/comps-foreman-el8.xml [hunk 5]
```diff
@@ -187,60 +338,201 @@
...
+      <packagereq type="default">rubygem-useragent-doc</packagereq>
+      <packagereq type="default">rubygem-validates_lengths_from_database-doc</packagereq>
+      <packagereq type="default">rubygem-webpack-rails-doc</packagereq>
       <packagereq type="default">rubygem-websocket-driver-doc</packagereq>
       <packagereq type="default">rubygem-websocket-extensions-doc</packagereq>
+      <packagereq type="default">rubygem-will_paginate-doc</packagereq>
+      <packagereq type="default">rubygem-wirb-doc</packagereq>
+      <packagereq type="default">rubygem-x-editable-rails-doc</packagereq>
+      <packagereq type="default">rubygem-xmlrpc-doc</packagereq>
+      <packagereq type="default">tfm-rubygem-m-doc</packagereq>
```
Comment (ekohl) [inline comps/comps-foreman-el8.xml:535]: This looks odd on EL8.

**Comment by ehelms** [comment]
This seems to include puppet-agent-* packages that have already been built and merged. This might need a rebase as part of next push.

File: comps/comps-foreman-el8.xml (hunk 5 above)
Comment (zjhuntin) [reply comps/comps-foreman-el8.xml]: This was an artifact from using the add_gem_package script to sort for me. It's been removed.

**Comment by zjhuntin** [comment]
@ehelms rebased, @ekohl addressed that issue
